drift/lib/src/sqlite3/database.dart

```diff
@@ -17,15 +17,18 @@ abstract class Sqlite3Delegate<DB extends CommonDatabase>
   bool _hasCreatedDatabase = false;
   bool _isOpen = false;
 
+  final bool _syncPersistence;
```

Comment:
I'm not sure if this even needs to be optional? I think users expect changes to be in the database after awaiting the future.